assets/src/components/detours/detourRouteSelectionPanel.tsx

```diff
@@ -9,19 +9,21 @@ import {
 } from "../../schedule"
 import RoutePropertiesCard from "../mapPage/routePropertiesCard"
 
-interface SelectedRouteInfoWithRoute {
+interface SelectedRouteInfoWithRoutePatterns {
   selectedRoute: Route
   routePatterns: ByRoutePatternId<RoutePattern>
   selectedRoutePatternId: RoutePatternId | null
 }
 
-interface SelectedRouteInfoWithoutRoute {
-  selectedRoute: null
+interface SelectedRouteInfoWithoutRoutePatterns {
+  selectedRoute: null | Route
```

Comment:
question(non-blocking): Would this be a chance to convert this also to `undefined`?

Reply:
I'm happy to do that, but I have a vague memory of:
- `undefined` is an implicit missing, like data might just not be initialized yet, or might just be an enhancement that doesn't really _mean_ anything
- `null` is a semantic "missing", in the sense that the user has selected a "missing" option, or cleared an input.

So my sense was that `null` was actually correct here.

Like I said though - happy to correct this if I misunderstood.

Reply:
I'm good with that rationale 👍🏻 thanks for sharing your thought process!